File: Inventory/Corp.ERP.Inventory.Application/DTO/EquipmentDto.cs
using Corp.ERP.Inventory.Domain.Models;

namespace Corp.ERP.Inventory.Application.Dto;

public class EquipmentDto
{
    public virtual int Id { get; set; }
    public virtual string Name { get; set; }
    public virtual string Description { get; set; }
    public virtual string Code { get; set; }
    public virtual Boolean IsInUse { get; set; }
    public virtual DateTime? StartDateUsage { get; set; }
    public virtual UserDto? UsedBy { get; set; }
    public virtual StorageDto StorageUnit { get; set; }

    public static implicit operator EquipmentDto(Equipment _model)
    {
        return new EquipmentDto
        {
            Id = _model.Id,
            Name = _model.Name,
            Description = _model.Description,
            Code = _model.Code,
            IsInUse = _model.IsInUse,
            StorageUnit = _model.StorageUnit,
            StartDateUsage = _model.StartDateUsage,
            UsedBy = _model.UsedBy,
        };
    }
}

File: Presentation/Corp.ERP.Presentation.Web/src/app/inventory/application/equipment/DTOs/equipment.dto.ts
import { StorageUnitDTO } from "./storage-unit.dto";
import { UserDTO } from "./user.dto";

export interface EquipmentDTO {
  id: string;
  name: string;
  description: string;
  code: string;
  isInUse: boolean;
  startDateUsage: Date;
  usedBy: UserDTO;
  storageUnit: StorageUnitDTO;
}
